src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/MålgruppeValidering.kt

```diff
@@ -8,62 +8,65 @@ import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.MålgruppeType
-import kotlin.collections.contains
 
 object MålgruppeValidering {
     fun validerKanLeggeTilMålgruppeManuelt(
         stønadstype: Stønadstype,
         målgruppeType: MålgruppeType,
     ) {
-        val gyldig =
-            when (stønadstype) {
-                Stønadstype.BARNETILSYN, Stønadstype.LÆREMIDLER, Stønadstype.BOUTGIFTER ->
-                    when (målgruppeType) {
-                        MålgruppeType.AAP -> true
-                        MålgruppeType.DAGPENGER -> false
-                        MålgruppeType.NEDSATT_ARBEIDSEVNE -> true
-                        MålgruppeType.OMSTILLINGSSTØNAD -> true
-                        MålgruppeType.OVERGANGSSTØNAD -> true
-                        MålgruppeType.UFØRETRYGD -> true
-                        MålgruppeType.SYKEPENGER_100_PROSENT -> true
-                        MålgruppeType.INGEN_MÅLGRUPPE -> true
-                        MålgruppeType.TILTAKSPENGER -> false
-                        MålgruppeType.KVALIFISERINGSSTØNAD -> false
-                    }
-
-                Stønadstype.DAGLIG_REISE_TSO ->
-                    when (målgruppeType) {
-                        MålgruppeType.AAP -> true
-                        MålgruppeType.DAGPENGER -> false
-                        MålgruppeType.NEDSATT_ARBEIDSEVNE -> true
-                        MålgruppeType.OMSTILLINGSSTØNAD -> true
-                        MålgruppeType.OVERGANGSSTØNAD -> true
-                        MålgruppeType.UFØRETRYGD -> true
-                        MålgruppeType.SYKEPENGER_100_PROSENT -> false
-                        MålgruppeType.INGEN_MÅLGRUPPE -> true
-                        MålgruppeType.TILTAKSPENGER -> false
-                        MålgruppeType.KVALIFISERINGSSTØNAD -> false
-                    }
-
-                Stønadstype.DAGLIG_REISE_TSR ->
-                    when (målgruppeType) {
-                        MålgruppeType.INGEN_MÅLGRUPPE -> true
-                        MålgruppeType.TILTAKSPENGER -> true
-                        MålgruppeType.KVALIFISERINGSSTØNAD -> true
-                        MålgruppeType.AAP -> false
-                        MålgruppeType.DAGPENGER -> true
-                        MålgruppeType.NEDSATT_ARBEIDSEVNE -> false
-                        MålgruppeType.OMSTILLINGSSTØNAD -> false
-                        MålgruppeType.OVERGANGSSTØNAD -> false
-                        MålgruppeType.UFØRETRYGD -> false
-                        MålgruppeType.SYKEPENGER_100_PROSENT -> false
-                    }
-            }
-        feilHvisIkke(gyldig) {
+        feilHvisIkke(kanMålgruppeBrukesForStønad(stønadstype, målgruppeType)) {
             "målgruppe=$målgruppeType er ikke gyldig for $stønadstype"
         }
     }
 
+    fun kanMålgruppeBrukesForStønad(
+        stønadstype: Stønadstype,
+        målgruppeType: MålgruppeType,
+    ): Boolean =
+        when (stønadstype) {
+            Stønadstype.BARNETILSYN, Stønadstype.LÆREMIDLER, Stønadstype.BOUTGIFTER ->
```

Comment:
Man kunne evt flyttet denne logikken nærmere domeneobjektet. Tenkte først til `Stønadstype`, men den ligger jo i en egen modul, så det blir litt knotete. Går jo forøvrig som en extension-function.

Men det kunne jo vært mulig med en `målgruppeType.erGyldigFor(stønadstype)` 🤷